src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -1075,231 +1075,6 @@ class npc_scarlet_ghoul : public CreatureScript
 
 };
 
-/*####
-## npc_scarlet_miner_cart
-####*/
-
-enum ScarletMinerCart
-{
-    SPELL_CART_CHECK        = 54173,
-    SPELL_SUMMON_CART       = 52463,
-    SPELL_SUMMON_MINER      = 52464,
-    SPELL_CART_DRAG         = 52465,
-
-    NPC_MINER               = 28841
-};
-
-class npc_scarlet_miner_cart : public CreatureScript
-{
-    public:
-        npc_scarlet_miner_cart() : CreatureScript("npc_scarlet_miner_cart") { }
-
-        struct npc_scarlet_miner_cartAI : public PassiveAI
-        {
-            npc_scarlet_miner_cartAI(Creature* creature) : PassiveAI(creature)
-            {
-                me->SetDisplayId(me->GetCreatureTemplate()->Modelid1); // Modelid2 is a horse.
-            }
-
-            void JustSummoned(Creature* summon) override
-            {
-                if (summon->GetEntry() == NPC_MINER)
-                {
-                    _minerGUID = summon->GetGUID();
-                    summon->AI()->SetGUID(_playerGUID);
-                }
-            }
-
-            void SummonedCreatureDespawn(Creature* summon) override
-            {
-                if (summon->GetEntry() == NPC_MINER)
-                    _minerGUID.Clear();
-            }
-
-            void DoAction(int32 /*param*/) override
-            {
-                if (Creature* miner = ObjectAccessor::GetCreature(*me, _minerGUID))
-                {
-                    me->SetWalk(false);
-
-                    // Not 100% correct, but movement is smooth. Sometimes miner walks faster
-                    // than normal, this speed is fast enough to keep up at those times.
-                    me->SetSpeedRate(MOVE_RUN, 1.25f);
```

Comment:
Is the movementspeed ingame okay? Since we delete this, we now should get creatures default speed from sai.
SMART_ACTION_SET_MOVEMENT_SPEED (136) is useable now, isnt it?

Reply:
Fix the speed in creature_template

Reply:
I didn't notice any problems with speed.